fsspec/implementations/sftp.py

```diff
@@ -78,34 +78,46 @@ def rmdir(self, path):
         self.ftp.rmdir(path)
 
     def info(self, path):
-        s = self.ftp.stat(path)
-        if S_ISDIR(s.st_mode):
+        stat = self._decode_stat(self.ftp.stat(path))
+        stat["name"] = path + "/" if stat["type"] == "directory" else path
```

Comment:
Isn't "/" also added in `_decode_stat`?

Reply:
The thing here is that I didn't want to duplicate all the `_decode_stat` functionality. However, the path name in one case (`info`) is only retrieved from the `info` argument, and in the case of `ls` is retrieved from the `stat` object (only the filename, the complete path needs to be added with the `ls` path argument)

The way to handle is a little bit dirty, I pass the "parent_path" to construct the name there, otherwise it is done "outside" (for info)

Reply:
So long as we don't end up with "//".

Reply:
By the way, is there a convention that directories end with "/"? (I just followed the existing code)

I am just asking because of other modules do explicitly omit the final '/' (os.path.normpath or pathlib)

Reply:
Now you mention it, since it already says "directory", it probably makes sense to drop it, and that also makes the code here simpler.